src/ui/win32/bindings/MemoryViewerControlBinding.cpp

```diff
@@ -231,6 +231,42 @@ bool MemoryViewerControlBinding::HandleNavigation(UINT nChar)
             }
             return true;
 
+        case VK_ADD:
+        case VK_SUBTRACT:
+        {
+            auto nAddress = m_pViewModel.GetAddress();
+            const auto& pEmulatorContext = ra::services::ServiceLocator::Get<ra::data::context::EmulatorContext>();
+            auto iMem = pEmulatorContext.ReadMemory(m_pViewModel.GetAddress(), m_pViewModel.GetSize());
+            auto iMaxValue = (1ULL << (ra::data::MemSizeBytes(m_pViewModel.GetSize()) * 8)) - 1;
+            auto iModifier = 0x1;
+
+            if (bControlHeld)
+                iModifier *= 4;
+            if (bShiftHeld)
+                iModifier *= 8;
+
+            // Min and Max tresholds
+            if ((iMem >= iMaxValue and nChar == VK_ADD) or (iMem == 0x0 and nChar == VK_SUBTRACT))
+                return false;
+
+            // Set value to min or max when tresholds are exceeded
+            if ((iMaxValue - iMem < iModifier) and nChar == VK_ADD)
+                iModifier = (iMaxValue - iMem);
+            else if ((iMem < iModifier) and nChar == VK_SUBTRACT)
+                iModifier = iMem;
+
+            nChar == VK_SUBTRACT ? iMem -= iModifier : iMem += iModifier;
+            std::wstring sValueText = ra::data::MemSizeFormat(iMem, m_pViewModel.GetSize(), MemFormat::Hex);
+
+            // Writing every byte separately considerably improves stability
+            for (int i = sValueText.length(); i != 0; i -= 2)
+            {
+                std::wstring sByteValue = sValueText.substr(i - 2, 2);
+                pEmulatorContext.WriteMemoryByte(nAddress++, std::stoi(sByteValue, 0, 16));
+            }
+            return true;
```

Comment:
I think most of this code should be moved into the viewmodel and have unit tests written. I see something along the lines of:
```
m_pViewModel.AdjustCurrentValue(nChar == VK_SUBTRACT ? -iModifier : iModifier);
```